### DISCUSSION TITLE
Troubleshooting Pod Startup Issues in Digit Deployment

### DISCUSSION THREAD

**Discussion by dhwani-ankit** [egovernments/Digit-Core#123] [Announcements]
Hey,

We are currently deploying Digit on our AWS server, following the instructions in the document. On our local machine VM, we encountered an issue where some pods were unable to start. Please refer to the attached screenshot for more details. Upon investigation, we found that there is an issue with the git-sync container.

<img width="649" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/b93c03be-a855-439c-b934-d1e56bfbd2cd">

<img width="960" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/188079f5-a25a-49c7-8d71-eb9d38f3ba7c">

Additionally, although we can access the frontend, we are unable to fill in the section labeled "CS_SELECT_CITY_CHOOSE_CITY".

<img width="951" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/b3fb487a-eebe-4af1-a0a7-f09d7a9f55ed">

Furthermore, when we made a request using postman to create a user, we encountered the following issue:

<img width="960" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/d6638242-028b-4e40-bf3a-8209a7be0e9b">

Any insights, suggestions, or guidance would be greatly appreciated.

**Comment by SriPadma8997-egov** [comment]
Hi @dhwani-ankit 
Please make sure you have added public key in github account which has access to the MDMS repo and the corresponding private key in env-secrets.yaml file. And also make sure you have added the latest known-host in env-secrets.yaml ([ref)](https://github.com/egovernments/DIGIT-DevOps/blob/c47fae4ab940683da767b77cee463f0891bb1d5a/config-as-code/environments/egov-demo-secrets.yaml#L55)

Please refer to the steps 3 & 4 in this [doc](https://core.digit.org/guides/installation-guide/quick-setup/digit-installation-on-aws)

**Reply by dhwani-ankit** [reply]
Hey,

I have generated new keys and added them to the file [/home/ubuntu/DIGIT-DevOps/deploy-as-code/helm/environments/quickstart-config-secrets.yaml].

<img width="448" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/334099c2-d1a2-4332-9e4a-ba2865d39c0b">

We are currently working on the "quickstart" branch.

However, we are still encountering the same issue.

<img width="960" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/afebdba0-8904-4a40-9de8-d8070fc20f06">

However, I can clone the same repo directly on the remote using same ssh key.

**Reply by SriPadma8997-egov** [reply]
Please use DIGIT-Bootcamp branch instead of quickstart
Ref to this [doc](https://core.digit.org/guides/installation-guide/quick-setup/digit-installation-on-aws) to deploy DIGIT.

**Reply by SriPadma8997-egov** [reply]
And also the private key should start with BEGIN RSA PRIVATE KEY not OPENSSH ([ref](https://github.com/egovernments/DIGIT-DevOps/blob/c47fae4ab940683da767b77cee463f0891bb1d5a/config-as-code/environments/egov-demo-secrets.yaml#L38))

**Reply by dhwani-ankit** [reply]
Thank You, The keys part should work now as all containers are running.

However, we're facing issues logging into the portal using the provided credentials. Additionally, we tried importing the seed from our local Postman to the application running on the AWS server.

<img width="367" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/34f9bc3f-c1b4-454a-97fc-f403bfd8deae">

Here's an example endpoint: POST: http://<public-ip>/user/users/_createnovalidate

<img width="900" alt="image" src="https://github.com/egovernments/Digit-Core/assets/124347605/5bc59494-e892-49df-afb5-5519ac858376">

**Reply by SriPadma8997-egov** [reply]
Are you using DIGIT-Bootcamp branch orelse quickstart branch

**Reply by dhwani-ankit** [reply]
Currently using quickstart branch, as we are setting the same using deploy-as-code

**Reply by SriPadma8997-egov** [reply]
Use DIGIT-Bootcamp branch , quickstart uses old release charts, it may not work properly now.

**Reply by dhwani-ankit** [reply]
We would like to set up the application on a single server [Demo Environment]. We have followed your documentation for the mentioned branch, but it seems to be using EKS and creating multiple servers, which is not our desired setup.

We noticed that **DIGIT-Bootcamp** branch utilizes **infra-as-code**, and for demo purposes, we believe it would not be a suitable option. If there are specific steps available for **deploying as code** using the same branch, we kindly request you to share them with us. 

This information would be immensely helpful in achieving our goal.

**Reply by shashwat-egov** [reply]
Hi, do you want to deploy all DIGIT services on a single server(node) itself?

**Reply by dhwani-ankit** [reply]
Yes, I want to deploy everything on a single server.

**Reply by shashwat-egov** [reply]
Currently, deploying DIGIT backbone services, core services and then any solution like PGR takes about 3 nodes of the type r5ad.large with the following specs - 2 vCPU, 16 gigs of memory, 1 x 75 NVMe SSD storage, Up to 10 GBPS network bandwidth. So, if you use an instance type which can provide enough CPU, memory and storage, you can set the number of instances to 1 and deploy it on AWS. You can change the number of nodes here - https://github.com/egovernments/DIGIT-DevOps/blob/c47fae4ab940683da767b77cee463f0891bb1d5a/infra-as-code/terraform/sample-aws/variables.tf#L45

**Comment by dhwani-ankit** [comment]
The setup is complete, and we can now log in to the same branch. Can you assist us with a new query regarding how to reflect our changes on the system?" Ref: [view](https://github.com/egovernments/DIGIT-DevOps/discussions/1817)
